### PR TITLE
Added missing methods and fixed existing ones with file download in job artifacts API

### LINKED ISSUE #586: Download a single artifact file by job ID
Feature request: https://docs.gitlab.com/ee/api/job_artifacts.html#download-a-single-artifact-file-by-job-id

Currently, the library allows to download all artifacts in gzip format only.

### PR DESCRIPTION
**Pull request by akashdotsrivastava**
Closes #586 

* Fixes the existing file download methods to correctly serve `FileResponse`
* Adds missing methods for downloading single file from artifacts for a job, and for a branch/tag from https://docs.gitlab.com/ee/api/job_artifacts.html#get-job-artifacts

### CODE REVIEW

**Comment by akashdotsrivastava** [comment]
@NARKOZ 
Hi. If it's okay, would you mind adding the `hacktoberfest-accepted` label to this PR before you approve, if/when you approve? Would be great if this becomes a part of my hacktoberfest contributions 😊

**Review by NARKOZ — APPROVED** [review]

**Comment by NARKOZ** [comment]
@akashdotsrivastava sure. Thanks for the contribution.

**Pull request merged**
